index.js

```diff
@@ -391,6 +393,12 @@ function wrapPromise() {
     return function wrappedThen() {
       var promise = this;
       var next = original.apply(promise, Array.prototype.map.call(arguments, bind));
+
+      next.__asl_wrapper = function proxyWrapper(ctx, fn, val, last) {
+        promise.__asl_wrapper(ctx, function () {}, null, next)
+        return next.__asl_wrapper(ctx, fn, val, last)
```

Comment:
If it works, it works, I guess. :see_no_evil: :hear_no_evil: :speak_no_evil: 

Unless you have a better idea, I think we can go with this.